experimenter/manifesttool/tests/test_github_api.py

```diff
@@ -158,3 +161,23 @@ def test_get_tags(self):
         self.assertEqual(rsps["2"].call_count, 1)
         self.assertEqual(rsps["3"].call_count, 1)
         self.assertEqual(rsps["4"].call_count, 0)
+
+    @responses.activate
+    def test_fetch_file_download(self):
+        """Testing github_api.fetch_file."""
+        responses.get(f"{GITHUB_RAW_URL}/repo/ref/file/path.txt", body=b"hello, world\n")
+
+        with TemporaryDirectory() as tmp_dir:
+            tmp_filename = Path(tmp_dir, "file.txt")
+
+            github_api.fetch_file("repo", "file/path.txt", "ref", tmp_filename)
+
+            self.assertTrue(tmp_filename.exists())
+
+            with tmp_filename.open("rb") as f:
+                contents = f.read()
+
+            self.assertEqual(contents, b"hello, world\n")
+
+        contents = github_api.fetch_file("repo", "file/path.txt", "ref")
+        self.assertEqual(contents, "hello, world\n")
```

Comment:
Why do we do these calls/assertions again outside of the `with` block?

Reply:
The calls are different. In the `with` block we are downloding a file, reading it, and comparing it

outside the with block, we are calling it without a download path and it returns the contents as as string.